packages/firebase_core/lib/src/firebase_app.dart

```diff
@@ -49,13 +49,16 @@ class FirebaseApp {
   /// Returns a list of all extant FirebaseApp instances, or null if there are
   /// no FirebaseApp instances.
   static Future<List<FirebaseApp>> allApps() async {
-    final List<Map<String, dynamic>> result = await channel.invokeMethod(
+    final List<dynamic> result = await channel.invokeMethod(
       'FirebaseApp#allApps',
     );
-    return result?.map((Map<String, dynamic> app) {
+    return result?.map<FirebaseApp>((dynamic entry) {
+      final Map<dynamic, dynamic> map = entry;
+      final String name = map['name'];
```

Comment:
Maybe this can just be `entry['name']`?

Not sure it is so much nicer though...